crawler/helpers/cherrypicked_samples.py

```diff
@@ -0,0 +1,199 @@
+import logging
+from typing import Dict, Iterator, List, Optional, Set, Tuple
+
+import pandas as pd
+import sqlalchemy
+from pandas import DataFrame
+
+from crawler.constants import FIELD_PLATE_BARCODE, FIELD_ROOT_SAMPLE_ID
+from crawler.sql_queries import SQL_MLWH_GET_CP_SAMPLES, SQL_MLWH_GET_CP_SAMPLES_BY_DATE
+from crawler.types import Config, SampleDoc
+
+logger = logging.getLogger(__name__)
+
+
+def extract_required_cp_info(samples: List[SampleDoc]) -> Tuple[Set[str], Set[str]]:
+    root_sample_ids = set()
+    plate_barcodes = set()
+
+    for sample in samples:
```

Comment:
This could be done using set comprehension.

Reply:
We could do, but would it require looping over the samples list twice? Unless we're missing something

Reply:
Yeah, ignore my comment, you are building two distinct sets which would require you to loop over samples twice if using comprehension.